examples/natmod/stm32wb55_transparent_vcp/_stm32wb55_transparent.c

```diff
@@ -0,0 +1,639 @@
+#include <string.h>
+#include "py/dynruntime.h"
+#include "py/stream.h"
+#include "stm32wbxx_ll_utils.h"
+#include "stm32wbxx_ll_system.h"
+#include "ports/stm32/mpbthciport.h"
+#include "ports/stm32/rfcore.h"
+
+// Based on STM32CubeWB\Projects\P-NUCLEO-WB55.USBDongle\Applications\BLE\BLE_TransparentModeVCP
```

Comment:
Are there any licensing/copyright issues to worry about here?  Was any code copied?

Reply:
Thanks for the reminder:

Reply:
As I suggested in previous comment I've split out all the re-used ST code into a separate C file which references a copy of the matching ST licence file. 

While it would certainly be possibly to re-implement the functionality from this file (it's just accessing a bunch of CPU registers really) I think this module is _most_ useful as an example of being able to directly re-use vendor code when suitable functions are provided this way.

Reply:
> Point 5 in particular appears to be a bit sticky

Yes it does.  It says the code cannot be used in a way that it would be subject to Open Source Terms, and in particular it mentions the MIT license.  I take this to mean that the MIT license cannot be made to apply to this ST code.  That makes sense, but I'm not sure if including the code in this repo (which has an MIT license at the top level) violates that point (5).

The code in this file is still under the ST license, we don't intend to change it to MIT.

I'm really not sure here.

Reply:
If we do go ahead with keeping this code, IMO the license text should just be pasted at the top of the ST file (and not in a separate ST_LICENSE file).

Reply:
After some discussion with more knowledgeable people on this topic, it's too dangerous to include this code with this license.  I think the main intent of this part of the license is to prevent it being incorporated in Linux as a driver, and I think it's safest if we just don't include it in this repo at all.

(That said, don't close this PR just yet!)

Reply:
Yeah it's a pretty fuzzy distinction going on in that licence. The writing there would definitely block usage within a gpl project as they enforce that everything is gpl compatible. Usage in something mit based might be different as mixed licencing is allowed there... but I definitely don't want to get your into hot water here!

This example would likely be safer split out into a separate repo, acting as a useful guide on how to have out-of-tree C module builds, perhaps then with a handy main.py file to set up the wb55 board/dongle as external radio for Unix micropython ble...